tests/components/SignUpForm.spec.js

```diff
@@ -0,0 +1,186 @@
+import React from 'react';
+import SignUpForm from 'components/SignUpForm';
+import { fireEvent, wait } from '@testing-library/react-native';
+
+import { renderWithRedux, configureStore } from '../helpers';
+
+describe('<SignUpForm />', () => {
+  let wrapper;
+  let store;
+  const props = {
+    onSubmit: jest.fn(),
+  };
+
+  beforeEach(() => {
+    store = configureStore();
+    wrapper = renderWithRedux(<SignUpForm {...props} />, store);
+  });
+
+  describe('Email Input', () => {
+    let input;
+    beforeEach(() => {
+      input = wrapper.queryByTestId('email-input');
+    });
+
+    it('should display an email field', () => {
+      expect(input).toBeTruthy();
+    });
+
+    describe('when the email input is not valid', () => {
+      beforeEach(() => {
+        fireEvent.changeText(input, 'example');
+        fireEvent.blur(input);
+      });
+
+      it('should show a email is not valid error', () => {
+        expect(wrapper.queryAllByLabelText('form-error')).toHaveLength(1);
+        expect(wrapper.queryByText('Email is not a valid email')).toBeTruthy();
+      });
+    });
+
+    describe('when the email input is valid', () => {
+      beforeEach(() => {
+        fireEvent.changeText(input, 'example@rootstrap.com');
+        fireEvent.blur(input);
+      });
+
+      it('should show a email is not valid error', () => {
+        expect(wrapper.queryAllByLabelText('form-error')).toHaveLength(0);
+        expect(wrapper.queryByText('Email is not a valid email')).toBeNull();
+        expect(wrapper.queryByText("Email can't be blank")).toBeNull();
+      });
+    });
+
+    describe('when the email input is not present', () => {
+      beforeEach(() => {
+        fireEvent.blur(input);
+      });
+
+      it('should show a required error', () => {
+        expect(wrapper.queryAllByLabelText('form-error')).toHaveLength(1);
+        expect(wrapper.queryByText("Email can't be blank")).toBeTruthy();
+      });
+    });
+  });
```

Comment:
Is there a way to reuse this? it's the exact same code as in the LoginForm Spec

Reply:
I would prefer not to this. I've been reading about testing coding style and some authors agreed that is better to be more explicit on tests, if you have duplicate code that tests similar functionality is better to be more verbose about it. Extracting the logic can lead to false positives and can be more difficult to track.